src/_semaphore.ts

```
@@ -0,0 +1,276 @@
+					lastExecutionTime: Date.now(),
+					lastOutput: output,
+					sessionId,
+					pid: process.pid,
+					version,
+					semaphoreType: this.semaphoreType,
+				};
+
+				logger.debug(`Updated semaphore for session: ${sessionId}`);
+			},
+			catch: error => new Error(`Failed to update semaphore: ${String(error)}`),
+		})();
```

ESLint

[error] Unsafe call of a(n) `error` type typed value.

(ts/no-unsafe-call)